src/producer.rs

```diff
@@ -0,0 +1,84 @@
+use rust_arroyo::backends::kafka::config::KafkaConfig;
+use rust_arroyo::backends::kafka::producer::KafkaProducer;
+use rust_arroyo::backends::kafka::types::KafkaPayload;
+
+use rust_arroyo::types::{Topic, TopicOrPartition};
+
+use crate::types::{
+    CheckResult,
+};
+use rust_arroyo::backends::{Producer, ProducerError};
+
+#[derive(Debug, thiserror::Error)]
+pub enum ExtractCodeError {
+    #[error(transparent)]
+    SerializationError(#[from] serde_json::Error),
+    #[error(transparent)]
+    ProducerError(#[from] ProducerError),
+}
+
+pub async fn produce_checker_result(
+    result: CheckResult,
+    topic: Topic,
+    config: KafkaConfig,
```

Comment:
It might make sense for this to not have to accept the topic/config, and for us to have it set up somewhere

Reply:
Yeah I agree, we can just put this method on a struct and load it in there